app/Spotify/Track.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Spotify;
+
+/**
+ * @author Michael Phillips <michael.phillips@realpage.com>
+ */
+class Track
+{
+    private const SPOTIFY_URL = 'https://open.spotify.com/track';
+
+    /**
+     * @param string $id The Spotify Track ID.
+     */
+    private function __construct(string $id)
+    {
+        $this->id = $id;
+    }
+
+    /**
+     * Create a Track from the URL.
+     *
+     * @param string $url The Spotify URL.
+     */
+    public static function fromUrl(string $url) : self
+    {
+        self::validateUrl($url);
+        $id = self::parseUrl($url);
+
+        return new self($id);
+    }
+
+    /**
+     * Validate the URL.
+     *
+     * @param string $url
+     * @throws UnexpectedValueException If $url is not a Spotify Track URL.
+     * @return void
+     */
+    private static function validateUrl(string $url) : void
```

Comment:
Typically, I put all validation rules in my VO's constructor, so they automatically run when I call `return new self()`.

Reply:
Since my constructor is private, this method is called from the `fromUrl` factory method.  See line 27.